compendium/test_bundles/ManagedServiceAndFactoryBundle/src/ManagedServiceFactoryImpl.cpp

```diff
@@ -0,0 +1,55 @@
+#include "ManagedServiceFactoryImpl.hpp"
+
+#include <iostream>
+
+namespace cppmicroservices {
+namespace service {
+namespace cm {
+namespace test {
+
+TestManagedServiceFactoryImpl::~TestManagedServiceFactoryImpl() = default;
+
+void TestManagedServiceFactoryImpl::Updated(std::string const& pid, AnyMap const& properties) {
+    std::lock_guard<std::mutex> lk(m_updatedMtx);
+    if (properties.empty()) {
+        std::cout << "    Updated called for pid \"" << pid << "\" with empty properties" << std::endl;
+        m_updatedCallCount[pid] -= 1;
+    } else {
+        auto const incrementBy = cppmicroservices::any_cast<int>(properties.AtCompoundKey("anInt"));
+        std::cout << "    Updated called for pid \"" << pid
+                  << "\" with value: " << std::to_string(incrementBy) << std::endl;
+        m_updatedCallCount[pid] += incrementBy;
+    }
+}
+
+void TestManagedServiceFactoryImpl::Removed(std::string const& pid) {
+    std::lock_guard<std::mutex> lk(m_removedMtx);
+    std::cout << "    Removed called for pid \"" << pid << "\" with empty properties" << std::endl;
```

Comment:
Just wondering if it's a good idea to hold the lock during the cout operation?Maybe move this statement down and release the lock before it?

Reply:
These couts will be removed. The guard is necessary though (even if the cout were to stay) so that we don't have a race if multiple threads call Removed concurrently. If the guard was after the couts, a race condition would exist with respect to the cout. Thanks!